vision/src/autogluon/vision/detector/detector.py

```diff
@@ -0,0 +1,270 @@
+                                      f'free memory: {free_gpu_memory[:ngpus_per_trial]} ' +
+                                      'Try reducing "batch_size" if you encounter memory issues.')
+            # check if hyperparameters overwriting existing config
+            for k, v in hyperparameters.items():
+                if k in config:
+                    raise ValueError(f'Overwriting {k} = {config[k]} to {v} by hyperparameters is ambiguous.')
+            config.update(hyperparameters)
+        if scheduler_options is not None:
+            config.update(scheduler_options)
+        # verbosity
+        if log_level > logging.INFO:
+            logging.getLogger('gluoncv.auto.tasks.object_detection').propagate = False
+            for logger_name in ('SSDEstimator', 'CenterNetEstimator', 'YOLOv3Estimator', 'FasterRCNNEstimator'):
+                logging.getLogger(logger_name).setLevel(log_level)
+                logging.getLogger(logger_name).propagate = False
+        task = _ObjectDetection(config=config)
+        task._logger.setLevel(log_level)
+        task._logger.propagate = True
+        self._detector = task.fit(train_data, tuning_data, 1 - holdout_frac, random_state)
+        self._detector._logger.setLevel(log_level)
+        self._detector._logger.propagate = True
+        self._fit_summary = task.fit_summary()
+
+    def _validate_kwargs(self, kwargs):
+        """validate and initialize default kwargs"""
+        kwargs['holdout_frac'] = kwargs.get('holdout_frac', 0.1)
+        if not (0 < kwargs['holdout_frac'] < 1.0):
+            raise ValueError(f'Range error for `holdout_frac`, expected to be within range (0, 1), given {kwargs["holdout_frac"]}')
+        kwargs['random_state'] = kwargs.get('random_state', None)
+        kwargs['nthreads_per_trial'] = kwargs.get('nthreads_per_trial', None)
+        kwargs['ngpus_per_trial'] = kwargs.get('ngpus_per_trial', None)
+        if kwargs['ngpus_per_trial'] is not None and kwargs['ngpus_per_trial'] > 0:
+            detected_gpu = get_gpu_count()
+            if detected_gpu < kwargs['ngpus_per_trial']:
+                raise ValueError(f"Insufficient detected # gpus {detected_gpu} vs requested {kwargs['ngpus_per_trial']}")
+        # tune kwargs
+        hpo_tune_args = kwargs.get('hyperparameter_tune_kwargs', {})
+        hpo_tune_args['num_trials'] = hpo_tune_args.get('num_trials', 1)
+        hpo_tune_args['search_strategy'] = hpo_tune_args.get('search_strategy', 'random')
+        if not hpo_tune_args['search_strategy'] in ('random', 'bayesopt', 'grid'):
+            raise ValueError(f"Invalid search strategy: {hpo_tune_args['search_strategy']}, supported: ('random', 'bayesopt', 'grid')")
+        hpo_tune_args['max_reward'] = hpo_tune_args.get('max_reward', 0.9)
+        if hpo_tune_args['max_reward'] < 0:
+            raise ValueError(f"Expected `max_reward` to be a positive float number between 0 and 1.0, given hpo_tune_args['max_reward']")
```

Comment:
This does not print out the value of max_reward. Missing `{}`. Also, the if statement doesn't check for `> 1` as the error states.